firebase-firestore/src/main/java/com/google/firebase/firestore/encoding/MapEncoder.java

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.encoding;
+
+import java.util.Map;
+
+/**
+ * Provides an interface of (de)encoding an custom object into(from) a nested map of primitive
+ * types. This interface is registered as a component into FirestoreRegistrar. Other library, like
+ * firestore Kotlin extension library can concretely implement this interface, and provide this
+ * component at runtime.
+ *
+ * <p>For the deserialization process, this {@code MapEncoder} is going to be used by the following
+ * Java classes: {@code DocumentSnapshot}, {@code QueryDocumentSnapshot}, and {@code QuerySnapshot}.
+ * For the serialization process, this {@code MapEncoder} is going to be used by the following Java
+ * classes: {@code DocumentReference}, {@code Transaction} and {@code WriteBatch}.
```

Comment:
No need to mention this

Reply:
I have removed from here, and keeping this part at the Ktx subclass.